### ISSUE TITLE
fix string comparison

### ISSUE THREAD

**Issue by sreev** [apache/incubator-heron#3047]
```
ERROR: /Users/svaddi/SreeVaddi/source/github/sreev/incubator-heron/heron/schedulers/src/java/BUILD:178:1: Building heron/schedulers/src/java/libmesos-scheduler-java.jar (8 source files) failed (Exit 1)
heron/schedulers/src/java/org/apache/heron/scheduler/mesos/framework/LaunchableTask.java:70: error: [ProtoStringFieldReferenceEquality] Comparing protobuf fields of type String using reference equality
      if (resource.getName() == name && resource.getScalar().getValue() >= value) {
                             ^
    (see https://errorprone.info/bugpattern/ProtoStringFieldReferenceEquality)
  Did you mean 'if (resource.getName().equals(name) && resource.getScalar().getValue() >= value) {'?
heron/schedulers/src/java/org/apache/heron/scheduler/mesos/framework/LaunchableTask.java:98: error: [ProtoStringFieldReferenceEquality] Comparing protobuf fields of type String using reference equality
      if (resource.getName() == name) {
                             ^
    (see https://errorprone.info/bugpattern/ProtoStringFieldReferenceEquality)
  Did you mean 'if (resource.getName().equals(name)) {'?
INFO: Elapsed time: 471.929s, Critical Path: 143.31s
INFO: 850 processes: 744 local, 106 worker.
FAILED: Build did NOT complete successfully

```

**Comment by sreev** [comment]
attached the patch:

[issue-3047.patch.txt](https://github.com/apache/incubator-heron/files/2482143/issue-3047.patch.txt)

**Comment by Code0x58** [comment]
This was fixed/included in cbe1e424ea9
